# Enunciado

Natural gas having CH4 = 82.4%, C2H6 = 15.8%, and N2 = 1.8 by
volume is fired in a boiler using 25% excess air. Ambient temperature is 25°C and relative humidity is 80%. Perform combustion calculations and determine the flue gas analysis based on wet and dry basis.

## Datos conocidos

In [1]:
metano                      = 0.824
etano                       = 0.158
nitrogeno                   = 1.8

exceso_aire                 = 0.25
porcentaje_aire_teorico     = 1 + exceso_aire


humedad                     = 0.8
T_amb                       = 25 + 273.15        # K
presion_ambiente            = 101.325 # Kpa

### Balance teórico

$$ 0.824 * CH_{4} + 0.158 * C_{2} H_{6} + 1.8 * N_{2} + 1.25 * a_{teórica} * (O_{2} + 3.76 N_{2}) $$
$$ \rightarrow X * CO_{2} + Y * H_{2}O + 0.25 * a_{teórica} * O_{2} + Z * N_{2}$$


## Incógnitas 
- X
- Y
- Z
- $a_{teórica}$

In [4]:
# Balance Carbono
x = metano + 2 * etano

# Bance Hidrógeno
y = (4 * metano + 6 *  etano) /2

# Balance Oxigeno
a_the = (2*x + y) / 2

# Balance nitrógeno 
z = porcentaje_aire_teorico * a_the * 3.76


print(f"Balance de carbono: x = {x}")
print(f"Balance de hidrógeno: y = {y}")
print(f"Balance de oxígeno: a_the = {a_the:.3f}")
print(f"Balance de nitrógeno: z = {z:.2f}")

Balance de carbono: x = 1.14
Balance de hidrógeno: y = 2.122
Balance de oxígeno: a_the = 2.201
Balance de nitrógeno: z = 10.34


### Balance teórico seco

$$ 0.824 * CH_{4} + 0.158 * C_{2}  H_{6} + 1.8 * N_{2} + 1.25 * 2.201 * (O_{2} + 3.76 N_{2})  $$ 

$$ \rightarrow  1.14 * CO_{2} + 2.122 * H_{2}O + 0.25 * 2.201 * O_{2} + 10.34 * N_{2}$$


## Análisis humedad

In [9]:
presion_saturacion = 3.1698 #Kpa A t = 25 ºC
print(f"Presión de saturación: {presion_saturacion:.2f} Kpa")

presion_vapor_aire = humedad * presion_saturacion
print(f"Presión de vapor de agua en aire: {presion_vapor_aire:.2f} Kpa")

moles_aire = a_the * 4.76
print(f"Moles aire {moles_aire:.3f}")

Presión de saturación: 3.17 Kpa
Presión de vapor de agua en aire: 2.54 Kpa
Moles aire 10.477


$$ N_{v aire} = (\frac{P_{V aire}}{P_{total}}) * N_{total} $$

$$ a_{the} * 4.76 = 2.201 * 4.76 = 10.47$$
$$ N_{v aire} = (\frac{P_{V aire}}{P_{total}}) * (10.47 - N_{v aire}) $$

In [10]:
from sympy import symbols, Eq, solve

N_aire = symbols('N_aire')

eq = Eq(N_aire, (presion_vapor_aire / presion_ambiente) * (moles_aire - N_aire) )
N_aire = solve(eq, N_aire)[0]


print(f"moles de vapor liquido {N_aire:.2f}")

moles de vapor liquido 0.26


### Balance húmedo

$$ 0.824 * CH_{4} + 0.158 * C_{2}  H_{6} + 1.8 * N_{2} + 2.75 * (O_{2} + 3.76 N_{2})$$ 
$$ \rightarrow  1.14 * CO_{2} + 2.122 * H_{2}O + 0.55 * O_{2} + 10.55 * N_{2}$$

---

$$ 0.824 * CH_{4} + 0.158 * C_{2}  H_{6} + 1.8 * N_{2} + 2.75 * (O_{2} + 3.76 N_{2}) + 0.26 * H_{2}O  $$
$$  \rightarrow  1.14 * CO_{2} + 2.382 * H_{2}O + 0.55 * O_{2} + 10.55 * N_{2} $$